DeprecationWarning: __array__ implementation doesn't accept a copy keyword

After a few months of inactivity, this has popped up.

The CI is failing on running tests (example: https://github.com/lobis/geant4-python-application/actions/runs/11079627911/job/30788984515) on Python 3.10, 3.11, 3.12 due to this deprecation warning.

`FAILED tests/test_application.py::test_seed_single_thread - DeprecationWarning: __array__ implementation doesn't accept a copy keyword, so passing copy=False failed. __array__ must implement 'dtype' and 'copy' keyword arguments.`

I suspect this has to do with awkward but it may also have to do directly with a new numpy version (2.x, the project was initially built with only 1.x in mind).

--- a/src/geant4_python_application/files/datasets.py
+++ b/src/geant4_python_application/files/datasets.py
@@ -131,7 +131,7 @@ def _get_dataset_download_size(dataset: Dataset) -> int:
 
 def _get_total_download_size(datasets_to_download: list[Dataset] = datasets) -> int:
     with concurrent.futures.ThreadPoolExecutor(
-        max_workers=len(datasets_to_download)
+            max_workers=len(datasets_to_download)
     ) as executor:
         futures = [
             executor.submit(_get_dataset_download_size, dataset)
@@ -199,7 +199,7 @@ def install_datasets(show_progress: bool = True):
         return
 
     # check if the datasets are present in the corresponding Geant4 directory
-    if not bool(missing_datasets(os.environ["GEANT4_DATA_DIR"])):
+    if "GEANT4_DATA_DIR" in os.environ and not bool(missing_datasets(os.environ["GEANT4_DATA_DIR"])):
         # return
         ...
 
@@ -217,14 +217,14 @@ The following Geant4 datasets will be installed: {", ".join([f"{dataset.name}@v{
         )
 
     with tqdm(
-        total=_get_total_download_size(datasets_to_download),
-        desc="Downloading Geant4 datasets",
-        disable=not show_progress,
-        unit="B",
-        unit_scale=True,
+            total=_get_total_download_size(datasets_to_download),
+            desc="Downloading Geant4 datasets",
+            disable=not show_progress,
+            unit="B",
+            unit_scale=True,
     ) as pbar:
         with concurrent.futures.ThreadPoolExecutor(
-            max_workers=len(datasets_to_download)
+                max_workers=len(datasets_to_download)
         ) as executor:
             futures = [
                 executor.submit(_download_extract_dataset, dataset, pbar)
@@ -235,7 +235,7 @@ The following Geant4 datasets will be installed: {", ".join([f"{dataset.name}@v{
     if show_progress:
         total_size_gb = sum(
             fp.stat().st_size for fp in Path(data_directory()).rglob("*")
-        ) / (1024**3)
+        ) / (1024 ** 3)
         print(f"Geant4 datasets size on disk after extraction: {total_size_gb:.2f}GB")
 
 

--- a/src/geant4_python_application/files/datasets.py
+++ b/src/geant4_python_application/files/datasets.py
@@ -131,7 +131,7 @@ def _get_dataset_download_size(dataset: Dataset) -> int:
 
 def _get_total_download_size(datasets_to_download: list[Dataset] = datasets) -> int:
     with concurrent.futures.ThreadPoolExecutor(
-            max_workers=len(datasets_to_download)
+        max_workers=len(datasets_to_download)
     ) as executor:
         futures = [
             executor.submit(_get_dataset_download_size, dataset)
@@ -199,7 +199,9 @@ def install_datasets(show_progress: bool = True):
         return
 
     # check if the datasets are present in the corresponding Geant4 directory
-    if "GEANT4_DATA_DIR" in os.environ and not bool(missing_datasets(os.environ["GEANT4_DATA_DIR"])):
+    if "GEANT4_DATA_DIR" in os.environ and not bool(
+        missing_datasets(os.environ["GEANT4_DATA_DIR"])
+    ):
         # return
         ...
 
@@ -217,14 +219,14 @@ The following Geant4 datasets will be installed: {", ".join([f"{dataset.name}@v{
         )
 
     with tqdm(
-            total=_get_total_download_size(datasets_to_download),
-            desc="Downloading Geant4 datasets",
-            disable=not show_progress,
-            unit="B",
-            unit_scale=True,
+        total=_get_total_download_size(datasets_to_download),
+        desc="Downloading Geant4 datasets",
+        disable=not show_progress,
+        unit="B",
+        unit_scale=True,
     ) as pbar:
         with concurrent.futures.ThreadPoolExecutor(
-                max_workers=len(datasets_to_download)
+            max_workers=len(datasets_to_download)
         ) as executor:
             futures = [
                 executor.submit(_download_extract_dataset, dataset, pbar)
@@ -235,7 +237,7 @@ The following Geant4 datasets will be installed: {", ".join([f"{dataset.name}@v{
     if show_progress:
         total_size_gb = sum(
             fp.stat().st_size for fp in Path(data_directory()).rglob("*")
-        ) / (1024 ** 3)
+        ) / (1024**3)
         print(f"Geant4 datasets size on disk after extraction: {total_size_gb:.2f}GB")
 
 

--- a/tests/test_application.py
+++ b/tests/test_application.py
@@ -54,7 +54,7 @@ def test_seed_single_thread():
             8.09506886e-03,
             1.22345221e00,
         ]
-        energy = np.array(events.track.step.energy[0][0][0:5])
+        energy = np.asarray(events.track.step.energy[0][0][0:5])
         assert np.allclose(energy, reference_value, atol=1e-5)
         assert app.seed == 1100
 

